core/src/main/scala/config/implicits/datastax.scala

```diff
@@ -0,0 +1,151 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package freestyle.cassandra
+package config.implicits
+
+import java.net.{InetAddress, InetSocketAddress}
+import java.util.concurrent.Executor
+
+import cats.implicits._
+import classy.DecodeError.WrongType
+import classy.config._
+import classy.{DecodeError, Decoder, Read}
+import com.datastax.driver.core.ProtocolOptions.Compression
+import com.datastax.driver.core.policies._
+import com.datastax.driver.core._
+import com.typesafe.config.Config
+import config._
+
+object datastax {
+
+  import config.implicits._
+
+  implicit val contactPointListRead: Read[Config, ContactPoints] = {
+    def trav[T](
+        l: List[String],
+        f: (String) => Either[DecodeError, T],
+        apply: List[T] => ContactPoints): Either[DecodeError, ContactPoints] =
+      l.traverse(f).map(apply)
+
+    def inetAddressParser(s: String): Either[DecodeError, InetAddress] = {
+      import cats.syntax.either._
+      Either.catchNonFatal(InetAddress.getByName(s)).leftMap(_ => WrongType("X.X.X.X", Some(s)))
+    }
+
+    def inetSocketAddressParser(s: String): Either[DecodeError, InetSocketAddress] = {
+      val SockedAddress = "([^:]+):([0-9]+)".r
+      s match {
+        case SockedAddress(host, port) => Right(new InetSocketAddress(host, port.toInt))
+        case _                         => Left(WrongType("<hostName>:<port>", Some(s)))
+      }
+    }
+
+    read[List[String], ContactPoints] { list =>
+      trav[InetSocketAddress](list, inetSocketAddressParser, ContactPointWithPortList)
+        .recoverWith {
+          case _ =>
+            trav[InetAddress](list, inetAddressParser, ContactPointList)
+        } match {
+        case Right(t) => Decoder.const(t)
+        case Left(e)  => Decoder.fail(e)
+      }
+    }
+  }
+
+  implicit val executorRead                   = instanceRead[Executor]
+  implicit val addressTranslatorRead          = instanceRead[AddressTranslator]
+  implicit val authProviderRead               = instanceRead[AuthProvider]
+  implicit val loadBalancingRead              = instanceRead[LoadBalancingPolicy]
+  implicit val reconnectionPolicyRead         = instanceRead[ReconnectionPolicy]
+  implicit val retryPolicyRead                = instanceRead[RetryPolicy]
+  implicit val speculativeExecutionPolicyRead = instanceRead[SpeculativeExecutionPolicy]
+  implicit val sslOptionsRead                 = instanceRead[SSLOptions]
+  implicit val threadingOptionsRead           = instanceRead[ThreadingOptions]
+  implicit val timestampGeneratorRead         = instanceRead[TimestampGenerator]
+
+  implicit val hostDistances: List[(String, HostDistance)] =
```

Comment:
Is there a reason to prefer `List[(String, HostDistance)]` (and others) to `Map[String, HostDistance]`?

`map.get(value)` should be faster than `list.find(_._1 == value)` below in `stringListRead`..